agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -162,6 +162,9 @@ type DockerClient interface {
 	// should be provided for the request.
 	ListContainers(context.Context, bool, time.Duration) ListContainersResponse
 
+	// PingDocker returns the Ping response from Docker's SystemPing API
+	PingDocker(context.Context, time.Duration) PingResponse
```

Comment:
nit: this method should probably be just called `Ping` since it's already inside `dockerGoClient`.

Reply:
I think I'll call it `SystemPing` --
I wanted to differentiate it from a simple Ping as it is calling a dedicated Docker API.
The fancy name for the API in Docker's docs is "System Ping" https://docs.docker.com/engine/api/v1.41/#operation/SystemPing